test/plugins/WebOfTrust/util/SynchronousDelayedBackgroundJobTest.java

```diff
@@ -0,0 +1,86 @@
+package plugins.WebOfTrust.util;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.Assert.*;
+
+/**
+ * A real simple unit test for {@link SynchronousDelayedBackgroundJob}.
+ *
+ * @author bertm
+ */
+public class SynchronousDelayedBackgroundJobTest {
+    /** Concurrency canary. */
+    private AtomicBoolean wasConcurrent;
+    /** Runnable for the current test. */
+    private Runnable run;
+    /** Background job for the current test. */
+    private SynchronousDelayedBackgroundJob job;
+
+    @Before
+    public void setUp() {
+        wasConcurrent = new AtomicBoolean(false);
+        run = null;
+        job = null;
+    }
+
+    /**
+     * Asserts that none of the canaries have been set.
+     */
+    @After
+    public void checkCanaries() {
+        assertFalse(wasConcurrent.get());
+    }
+
+    /**
+     * Surrounds the given runnable with the concurrency canary boilerplate.
+     */
+    private Runnable withConcurrencyCanary(final Runnable r) {
+        return new Runnable() {
+            private final AtomicBoolean isRunning = new AtomicBoolean(false);
+            @Override
+            public void run() {
+                if (isRunning.getAndSet(true)) {
+                    wasConcurrent.set(true);
+                }
+                r.run();
+                isRunning.set(false);
+            }
+        };
+    }
+
+    /**
+     * Tests a simple self-scheduling job for lack of deadlock, reasonable timing and termination.
+     */
+    @Test
+    public void selfSchedulingTest() throws Exception {
+        final AtomicInteger runCount = new AtomicInteger(0);
+        Runnable run = withConcurrencyCanary(new Runnable() {
+            @Override
+            public void run() {
+                runCount.incrementAndGet();
+                job.triggerExecution();
```

Comment:
I coded this with a rather specific deadlock in mind, so I didn't bother at the time. I'll change it to `0` though to allow the canaries to trip if anything is wrong here (for which I have no evidence).